src/tiled/mapdocumentactionhandler.cpp

```diff
@@ -299,16 +302,54 @@ void MapDocumentActionHandler::selectAll()
     }
 }
 
+void MapDocumentActionHandler::selectInverse()
+{
+    if (!mMapDocument)
+        return;
+
+    Layer *layer = mMapDocument->currentLayer();
+    if (!layer)
+        return;
+
+    if (TileLayer *tileLayer = layer->asTileLayer()) {
+        QRegion all(tileLayer->x(), tileLayer->y(),
+                  tileLayer->width(), tileLayer->height());
+
+        QUndoCommand *command = new ChangeSelectedArea(mMapDocument, all - mMapDocument->selectedArea());
+        mMapDocument->undoStack()->push(command);
+    } else if (ObjectGroup *objectGroup = layer->asObjectGroup()) {
+        QList<MapObject*> allObjects = objectGroup->objects();
+        QList<MapObject*> selectedObjects = mMapDocument->selectedObjects();
+        QList<MapObject*> notSelectedObjects;
+
+        for (auto& mapObject : allObjects) {
```

Comment:
Please don't write a `&` here because you're creating a reference to a pointer. Just `auto` is enough.